src/include/wiredtiger.h.in

```diff
@@ -5707,6 +5729,46 @@ struct __wt_page_log {
     int (*terminate)(WT_PAGE_LOG *page_log, WT_SESSION *session);
 };
 
+/*!
+ * The interface implemented by applications to provide a key management service
+ * implementation.
+ *
+ * Applications register implementations with WiredTiger by calling
+ * WT_CONNECTION::set_key_management.
+ *
+ * @snippet ex_key_management.c WT_KEY_MANAGEMENT register
+ */
+struct __wt_key_management {
```

Comment:
I have removed the need for WT_KEY_MANAGEMENT_LOAD_ARGS and WT_KEY_MANAGEMENT_GET_ARGS. I thought they weren't necessary, let me know otherwise.

Reply:
I have also changed the API names, IMO this feels like a cleaner name for the API please review if you disagree